lib/iamspec/action/generic_action.rb

```diff
@@ -23,9 +38,51 @@ def to_s
       end
     end
 
+    def with_credentials(creds)
+      @creds = creds
+      self
+    end
+
     def with_resource(resource_arn)
       @resource_arns = [resource_arn]
       self
     end
+
+    def with_policy(policy)
+      @policy = policy
+      self
+    end
+
+    def with_userid(userid)
```

Comment:
`userid`, `sourcevpce` and `sourceip`  to me feel rather specific to S3. Was there a reason to put them into `GenericAction` class instead of `S3Action`?

I might put a `with_context_entry(type,context_entry)` method into `GenericAction` and use that in the more specialized actions like `S3Action`.